components/net/fetch/methods.rs

```diff
@@ -640,64 +640,56 @@ fn http_redirect_fetch(request: Rc<Request>,
     // Step 1
     assert_eq!(response.return_internal.get(), true);
 
-    // Step 3
-    // this step is done early, because querying if Location exists says
-    // if it is None or Some, making it easy to seperate from the retrieval failure case
+    // Step 2
     if !response.actual_response().headers.has::<Location>() {
         return Rc::try_unwrap(response).ok().unwrap();
     }
 
-    // Step 2
+    // Step 3
     let location = match response.actual_response().headers.get::<Location>() {
         Some(&Location(ref location)) => location.clone(),
-        // Step 4
         _ => return Response::network_error()
     };
-
-    // Step 5
     let response_url = response.actual_response().url.as_ref().unwrap();
     let location_url = response_url.join(&*location);
-
-    // Step 6
     let location_url = match location_url {
         Ok(url) => url,
         _ => return Response::network_error()
     };
 
-    // Step 7
+    // Step 4
+    //TODO implement return network_error if not HTTP(S)
+
+    // Step 5
     if request.redirect_count.get() >= 20 {
         return Response::network_error();
     }
 
-    // Step 8
+    // Step 6
     request.redirect_count.set(request.redirect_count.get() + 1);
 
-    // Step 9
-    request.same_origin_data.set(false);
-
     let same_origin = if let Origin::Origin(ref origin) = *request.origin.borrow() {
         *origin == request.current_url().origin()
     } else {
         false
     };
     let has_credentials = has_credentials(&location_url);
-
-    // Step 10
+    // Step 7
```

Comment:
The newline is missing intentionally because the previous lines are preparation for Step 7 and not used in Step 6.